e2e2/test/cases/nvidia/main_test.go

```diff
@@ -129,15 +139,162 @@ func checkNodeTypes(ctx context.Context, config *envconf.Config) (context.Contex
 	return ctx, nil
 }
 
+func createMetricManager(ctx context.Context, config *envconf.Config) (context.Context, error) {
+	log.Printf("AMP url is set to %s", *ampMetricUrl)
+	metadataLabel, err := getMetadataLabel(ctx, config)
+	if err != nil {
+		return ctx, err
+	}
+	metricManager = metric.NewMetricManager(metadataLabel, awsCfg, *ampMetricUrl, *ampMetricRoleArn)
+
+	return ctx, nil
+}
+
+func getMetadataLabel(ctx context.Context, cfg *envconf.Config) (map[string]string, error) {
+	restConfig := cfg.Client().RESTConfig()
+	// Initialize Kubernetes client from the provided REST config
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+
+	// List nodes in the cluster
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	// Ensure at least one node is available
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found in the cluster")
+	}
+
+	// Get instance type and metadata from the first node
+	node := nodes.Items[0]
+	osType := node.Status.NodeInfo.OSImage
+
+	// Get k8s cluster version
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf(" error in discoveryClient %v", err)
+	}
+	information, err := discoveryClient.ServerVersion()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get k8s version: %v", err)
+	}
+	k8sVersion := fmt.Sprintf("%s.%s", information.Major, information.Minor)
+
+	// get ami id
+	instanceId := regexp.MustCompile(`i-[a-zA-Z0-9]+`).FindString(node.Spec.ProviderID)
+	// describe the instance
+	ec2Client := ec2.NewFromConfig(awsCfg)
+	instance, err := ec2Client.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+		InstanceIds: []string{instanceId},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe instance: %v", err)
+	}
+	amiId := *instance.Reservations[0].Instances[0].ImageId
+
+	// Construct the metadata labels
+	metadataLabels := map[string]string{
+		"instance_type":      *nodeType,
+		"node_count":         fmt.Sprintf("%d", nodeCount),
+		"kubernetes_version": k8sVersion,
+		"efa_enabled":        fmt.Sprintf("%t", *efaEnabled),
+		"efa_count":          fmt.Sprintf("%d", efaPerNode),
+		"ami_id":             amiId,
+		"os_type":            osType,
+	}
+
+	// Create a job to fetch the logs of meta info
```

Comment:
seems like you could just log these details in your actual test run instead of using a separate pod to print them

Reply:
I couldn't, I tried to add `ENTRYPOINT` in my dockerfile, but the nccl test pods doesn't print these details.

Reply:
Wondering the same..the launcher pods or worker pods should have these details right as they also run the entrypoint script ?